// sforms/index.js

function sforms(){
  this.formList = {}
}

sforms.prototype.init =  function(context){
  var self = this
  forms = context.find('form[sform]')
  console.log(forms)
  forms.each(function(i, elem){
    var $elem = $(elem)
    var id = $elem.attr('sform')
    $elem.submit(function(e){
      e.preventDefault()
      self.formList[id].update().save()
    })
    self.formList[id] = new sform($elem)
  })
}

sforms.prototype.dumpall = function(){
  console.log(this.formList)
}

function sform ($form){
  var self = this
  self.$form = $form
  self.update = function(model){
    if(!model){
      model = {}
      $form.find('[name]').each(function(i, elem){
        var $elem = $(elem)
        var prop = $elem.attr('name')
        if(prop) model[prop] = $elem.val()
        self.model = model
      })
    }
    else self.model = model
    console.log(model)
    return self
  }
  self.save = function(){
    console.log('MMM',this.model)
    $.ajax({
      type:"POST",
      url: self.$form.attr('url'),
      data: self.model,
      complete: function(data){
        console.log('success!!!!')
        console.log(data)
      }
    })
  }
}

module.exports = function(){
  return new sforms()
}
