// _twitter/index.js
var request = require('request')
var qs = require('querystring')

function pocket(config){
  this.accessToken = config.accessToken
  this.consumerKey = config.consumerKey
}

pocket.prototype.getRequestToken = function (redirectUri, cb){
  var r = {
    url: 'https://getpocket.com/v3/oauth/request',
    form: {
      consumer_key: this.consumerKey,
      redirect_uri: redirectUri
    }
  }
  request.post(r, function(err, resp, body){
    if(err) cb(err)
    else{
      var data = qs.parse(body)
      cb(null, data)
    }
  })
}

pocket.prototype.getAccessToken = function(requestToken, cb){
  var r = {
    url: 'https://getpocket.com/v3/oauth/authorize',
    form: {
      consumer_key: this.consumerKey,
      code: requestToken
    }
  }
  request.post(r, function(err, resp, body){
    if(err) cb(err)
    else{
      var data = qs.parse(body)
      cb(null, data)
    }
  })
}

pocket.prototype.get = function(cb){
  r = {
    url: 'https://getpocket.com/v3/get',
    form: {
      consumer_key: this.consumerKey,
      access_token: this.accessToken
    }
  }
  request.post(r, function(err, resp, body){
    if(err) cb(err)
    else{
      var data = JSON.parse(body)
      cb(null, data)
    }
  })
}

module.exports = function(config){
  return new pocket(config)
}
